src/orion/core/utils/random_state.py

```diff
@@ -80,20 +80,26 @@ def seed(cls, base_seed: int | None) -> RandomState:
         return random_state
 
 
+class Algo(Protocol):
```

Comment:
Perhaps a more explicit name? Since this might eventually be used on things other than just algorithms?
```suggestion
class HasRandomState(Protocol):
```

Reply:
Agreed, done in a39c818.